tests/Unit/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Test_GaugeWave.cpp

```diff
@@ -0,0 +1,302 @@
+#include <string>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/Pypp/CheckWithRandomValues.hpp"
+#include "tests/Unit/Pypp/SetupLocalPythonEnvironment.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+struct GaugeWaveProxy : gr::Solutions::GaugeWave {
+  using gr::Solutions::GaugeWave::GaugeWave;
+
+  template <typename DataType>
+  using variables_tags = gr::Solutions::GaugeWave::tags<DataType>;
+
+  template <typename DataType>
+  tuples::tagged_tuple_from_typelist<variables_tags<DataType>> test_variables(
+      const tnsr::I<DataType, 3>& x, const double t) const noexcept {
+    return variables(x, t, variables_tags<DataType>{});
+  }
+};
+
+struct GaugeWaveOptionTag {
+  using type = gr::Solutions::GaugeWave;
+  static constexpr OptionString help{"A gauge-wave solution"};
+};
+
+template <typename DataType>
+tnsr::I<DataType, 3, Frame::Inertial> spatial_coords(
+    const DataType& used_for_size) noexcept {
+  auto x = make_with_value<tnsr::I<DataType, 3, Frame::Inertial>>(used_for_size,
+                                                                  0.0);
+  get<0>(x) = 1.32;
+  get<1>(x) = 0.82;
+  get<2>(x) = 1.24;
+  return x;
+}
+
+template <typename DataType>
+void test_gauge_wave(const gr::Solutions::GaugeWave& solution,
+                     const DataType& used_for_size) noexcept {
+  // The solution is as follows, with H = 1 - A \sin((2*\pi/d)*(x-t))
+  // and \partial_x H = - A (2*\pi/d) \cos((2*\pi/d)*(x-t)):
+  //
+  // \gamma_{ij} = 0, except \gamma_{xx} = H and \gamma_{yy} = \gamma_{zz} = 1
+  // \gamma^{ij} = 0, except \gamma^{xx} = H and \gamma^{yy} = \gamma^{zz} = 1
+  // \partial_t\gamma_{ij} = 0, except \partial_t\gamma_{xx} = - \partial_x H
+  // \partial_x\gamma_{ij} = 0, except \partial_x\gamma_{xx} = \partial_x H
+  // \sqrt{\gamma} = \sqrt{H}
+  // \alpha = \sqrt{H}
+  // \partial_t \alpha = - \partial_x H / (2 \sqrt{H})
+  // \partial_x \alpha = \partial_x H / (2 \sqrt{H})
+  // \partial_y \alpha = \partial_z \alpha = 0
+  // \beta^i = 0
+  // \partial_t \beta^i = 0
+  // \partial_i \beta^j = 0
+  // K_{ij} = 0, except K_{xx} = \partial_x H / (2 \sqrt{H})
+
+  // Parameters for GaugeWave solution
+  const double amplitude = 0.24;
+  const double wavelength = 4.4;
+
+  // Verify that the solution passed in was constructed with the correct
+  // parameters
+  REQUIRE(solution == gr::Solutions::GaugeWave{amplitude, wavelength});
```

Comment:
Should this be `CHECK`?

Reply:
No, it should be a `REQUIRE`. Everything after this doesn't make sense if this fails.